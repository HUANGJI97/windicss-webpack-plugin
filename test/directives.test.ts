import { describe, it, expect } from 'vitest'
import { getModuleSource, vueWebpackCompiler } from './helpers'

describe("Directives test", function() {

  it('should render vue', done => {
    const compiler = vueWebpackCompiler('vue-directives')
    compiler.run((err, stats) => {
      expect(stats.compilation.errors).toStrictEqual([])
      const windiComponents = getModuleSource('virtual:windi-components.css', stats)
      expect(windiComponents).toContain('.bar')
      expect(windiComponents).toMatchSnapshot('windi components')

      done(err)
    });
  });
});
