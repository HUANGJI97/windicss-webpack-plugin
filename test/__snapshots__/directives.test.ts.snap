// Vitest Snapshot v1

exports[`Directives test > should render vue > windi components 1`] = `
"// Imports
import ___CSS_LOADER_API_IMPORT___ from \\"../../../node_modules/.pnpm/css-loader@5.2.7_webpack@4.46.0/node_modules/css-loader/dist/runtime/api.js\\";
var ___CSS_LOADER_EXPORT___ = ___CSS_LOADER_API_IMPORT___(function(i){return i[1]});
// Module
___CSS_LOADER_EXPORT___.push([module.id, \\"/* windicss layer components */\\\\n.bar {\\\\n  --tw-bg-opacity: 1;\\\\n  background-color: rgba(252, 231, 243, var(--tw-bg-opacity));\\\\n}\\\\n@media (min-width: 768px) {\\\\n  .bar {\\\\n    background: red;\\\\n  }\\\\n}\\", \\"\\"]);
// Exports
export default ___CSS_LOADER_EXPORT___;
"
`;
